File: custom_services/src/service_client.cpp
#include "ros/ros.h"
#include "custom_services/MeanThreeInts.h"
#include <cstdlib>

int main(int argc, char **argv)
{
  ros::init(argc, argv, "service_client");
  if (argc != 4)
  {
    ROS_INFO("usage: simple_client X Y Z");
    return 1;
  }

  ros::NodeHandle n;
  ros::ServiceClient client = n.serviceClient<custom_services::MeanThreeInts>("mean_three_ints");
  custom_services::MeanThreeInts srv;
  srv.request.a = atoll(argv[1]);
  srv.request.b = atoll(argv[2]);
  srv.request.c = atoll(argv[3]);
  if (client.call(srv))
  {
    ROS_INFO("Mean: %f", srv.response.mean);
  }
  else
  {
    ROS_ERROR("Failed to call service");
    return 1;
  }

  return 0;
}

File: custom_services/scripts/service_client.py
#!/usr/bin/env python

import sys
import rospy
from custom_services.srv import *

def service_client(x, y, z):
    rospy.wait_for_service('mean_three_ints')
    try:
        mean_three_ints = rospy.ServiceProxy('mean_three_ints', MeanThreeInts)
        resp = mean_three_ints(x, y, z)
        return resp.mean
    except rospy.ServiceException as e:
        print("Service call failed: %s"%e)

def usage():
    return "%s [x y z]"%sys.argv[0]

if __name__ == "__main__":
    if len(sys.argv) == 4:
        x = int(sys.argv[1])
        y = int(sys.argv[2])
        z = int(sys.argv[3])
    else:
        print(usage())
        sys.exit(1)

    print("Requesting %s %s %s"%(x, y, z))
    print("(%s+%s+%s)/3 = %s"%(x, y, z, service_client(x, y, z)))
